### PR TITLE
Refactor(eos_cli_config_gen): Wildcard dict to list for `prefix_lists`

### PR DESCRIPTION
**Pull request by KonikaChaurasiya-GSLab**
## Data model key

`prefix_lists`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update all `host_vars` under molecule scenario `eos_cli_config_gen_v4.0` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Run molecule `cd ansible_collections/arista/avd ; molecule converge -s eos_cli_config_gen`
  - [x] Verify no changes to generated configs/docs
- [x] Run molecule `cd ansible_collections/arista/avd ; molecule converge -s eos_cli_config_gen_v4.0`
  - [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1 (@ClausHolbechArista):
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [ ] Verify that CI pass

- Reviewer 2:
  - [x] Verify data model changes
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ipv6-prefix-lists.j2 [hunk 1]
```diff
@@ -1,10 +1,12 @@
...
+{% for ipv6_prefix_list in ipv6_prefix_lists | arista.avd.convert_dicts('name') |
+arista.avd.natural_sort('name') %}
 !
-ipv6 prefix-list {{ ipv6_prefix_list }}
-{%     for sequence in ipv6_prefix_lists[ipv6_prefix_list].sequence_numbers | arista.avd.natural_sort %}
-{%         if ipv6_prefix_lists[ipv6_prefix_list].sequence_numbers[sequence].action is arista.avd.defined %}
-   seq {{ sequence }} {{ ipv6_prefix_lists[ipv6_prefix_list].sequence_numbers[sequence].action }}
+ipv6 prefix-list {{ ipv6_prefix_list.name }}
+{%     for sequence in ipv6_prefix_list.sequence_numbers | arista.avd.convert_dicts('sequence') |
+arista.avd.natural_sort('sequence') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ipv6-prefix-lists.j2]: I think this wrapped to a new line by mistake. Same elsewhere in this and the other files.

Comment (KonikaChaurasiya-GSLab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/ipv6-prefix-lists.j2]: Yes, I have corrected this. Thanks

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Review by emilarista — APPROVED** [review]
LGTM!

**Pull request merged**